cloud/deploy/deploy.go

```diff
@@ -996,3 +1002,89 @@ func DeployClientImage(deployInput InputClientDeploy) error { //nolint:gocritic
 
 	return nil
 }
+
+// validateClientImageRuntimeVersion validates that the client image runtime version
+// is not newer than the deployment runtime version
+func validateClientImageRuntimeVersion(deployInput InputClientDeploy, platformCoreClient astroplatformcore.CoreClient) error { //nolint:gocritic
+	// Skip validation if no deployment ID provided
+	if deployInput.DeploymentID == "" {
+		return nil
+	}
+
+	// Get current context for organization info
+	c, err := config.GetCurrentContext()
+	if err != nil {
+		return errors.Wrap(err, "failed to get current context")
+	}
+
+	// Get deployment information
+	deployInfo, err := getImageName(deployInput.DeploymentID, c.Organization, platformCoreClient)
```

Comment:
`getImageName` is a very confusing name for what the method is doing. Should we fix/update the name?